FogCreek/Glitch-Community
# Abstracted droplet: normalize project options

replaces #654.

## Links
* Remix link: https://abstracted-droplet.glitch.me/
* Manuscript link: https://glitch.manuscript.com/f/cases/3328765/Make-project-options-consistent-and-available-everywhere

## Changes:
* pass all possible project options as props, instead of going section by section
* pass profile object (e.g. user, team) instead of `isAuthorized` as project options
* provide default project options for addProjectToCollection, joinTeamProject, leaveProject, deleteProject that are accessible from everywhere

## How To Test:
* visit https://abstracted-droplet.glitch.me/
* verify project options are not available when logged out
* log in
* verify that pin / unpin / feature options are correct for your pinned vs recent projects
* verify that join / leave options are correct for your team projects
* verify that you cannot pin / unpin / feature / join / delete other users' or other teams' projects
